File: Sources/FeedbackMode.swift
import SwiftUI
import UIKit

// MARK: - Model

enum ExperimentStatus: String, Codable {
    case suggested, trialing, reverted, expired
}

/// A suggestion, and — when it maps to a presentation variant — its trial.
/// Trials carry a hard 7-day clock (docs/FEEDBACK-MODE.md).
struct Experiment: Identifiable, Codable, Equatable {
    let id: String
    var title: String
    var detail: String
    var surface: String
    var flag: String          // "" = suggestion only, no variant to try
    var status: ExperimentStatus
    var startedAt: Date?
    var expiresAt: Date?
    var sentAt: Date

    var daysLeft: Int? {
        guard status == .trialing, let end = expiresAt else { return nil }
        return max(0, Int(ceil(end.timeIntervalSince(Date()) / 86_400)))
    }

    var statusLine: String {
        switch status {
        case .suggested: return "Suggested · awaiting decision"
        case .trialing: return daysLeft.map { "Trial · \($0)d left on this device" } ?? "Trial"
        case .reverted: return "Trial ended by you · still in review"
        case .expired: return "Trial reached its week · still in review"
        }
    }

    static func load() -> [Experiment] {
        guard let data = UserDefaults.standard.data(forKey: "experiments.v1"),
              let decoded = try? JSONDecoder().decode([Experiment].self, from: data) else { return [] }
        return decoded
    }

    static func save(_ experiments: [Experiment]) {
        if let data = try? JSONEncoder().encode(experiments) {
            UserDefaults.standard.set(data, forKey: "experiments.v1")
        }
    }
}

/// The variants this build knows how to try. Presentation-only, always.
enum Variants {
    static let focusHome = "focus-home"
    static func matches(_ text: String, surface: String) -> String? {
        let folded = text.lowercased()
        let focusWords = ["focus", "minimal", "simpl", "hide", "quiet", "clean", "less"]
        if surface == "Home" || folded.contains("home"),
           focusWords.contains(where: { folded.contains($0) }) {
            return focusHome
        }
        return nil
    }
}

// MARK: - Floating bubble

/// The feedback door — only exists while the program is on AND this device
/// opted in. Quiet, corner, never over the session plane.
struct FeedbackBubble: View {
    @EnvironmentObject var store: AppStore
    @State private var chatOpen = false

    var body: some View {
        Button { chatOpen = true } label: {
            ZStack {
                Circle().fill(DT.heroGradient)
                ClineBotShape()
                    .fill(style: FillStyle(eoFill: true))
                    .foregroundStyle(.white)
                    .frame(width: 22, height: 22)
            }
            .frame(width: 46, height: 46)
            .shadow(color: DT.violet.opacity(0.4), radius: 10, y: 4)
        }
        .buttonStyle(Pressable())
        .accessibilityLabel("Design with Cline")
        .accessibilityHint("Suggest a feature — chat stays on this device")
        .sheet(isPresented: $chatOpen) {
            FeedbackChatView()
        }
    }
}

// MARK: - The design chat

private struct FBMessage: Identifiable {
    let id = UUID()
    let fromCline: Bool
    let text: String
}

/// Chat with Cline to shape a suggestion. The conversation is ephemeral —
/// in memory, gone when the sheet closes. Only the structured draft leaves,
/// and only on an explicit Send. (Cline here is a rule-based design
/// persona in the preview build; the honesty chrome says so.)
struct FeedbackChatView: View {
    @EnvironmentObject var store: AppStore
    @Environment(\.colorScheme) private var scheme
    @Environment(\.dismiss) private var dismiss

    @State private var messages: [FBMessage] = [
        FBMessage(fromCline: true,
                  text: "What should Drive do better? Describe it like you'd tell a friend — I'll shape it into a proposal you can send.")
    ]
    @State private var input = ""
    @State private var surface = "Home"
    @State private var draftTitle: String?
    @State private var draftSummary = ""
    @State private var sent = false

    private let surfaces = ["Home", "Work", "Agents", "Tasks", "Profile", "Artifacts"]

    private var matchedFlag: String? {
        Variants.matches(draftSummary, surface: surface)
    }

    var body: some View {
        NavigationStack {
            VStack(spacing: 0) {
                honesty.padding(.top, 10)
                ScrollView {
                    VStack(alignment: .leading, spacing: 10) {
                        ForEach(messages) { message in
                            bubble(message)
                        }
                        if draftTitle != nil { draftCard.padding(.top, 6) }
                    }
                    .padding(.horizontal, 18)
                    .padding(.top, 14)
                }
                surfaceChips
                inputBar
            }
            .background(DT.page(scheme).ignoresSafeArea())
            .navigationTitle("Design with Cline")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .cancellationAction) {
                    Button("Close") { dismiss() }
                }
            }
        }
        .presentationDetents([.large])
        .presentationCornerRadius(22)
    }

    private var honesty: some View {
        HStack(spacing: 7) {
            Circle().fill(DT.live(scheme)).frame(width: 6, height: 6)
            Text("CHAT STAYS ON THIS DEVICE · ONLY WHAT YOU SEND LEAVES")
                .tracking(0.8)
                .scaledFont(8.5, .bold)
                .foregroundStyle(DT.ink55(scheme))
        }
        .padding(.horizontal, 12).padding(.vertical, 7)
        .background(DT.surface2(scheme))
        .clipShape(Capsule())
    }

    private func bubble(_ message: FBMessage) -> some View {
        HStack(alignment: .bottom, spacing: 8) {
            if message.fromCline {
                AvatarChip(letter: "C", color: DemoData.coder, size: 24)
            } else {
                Spacer(minLength: 40)
            }
            Text(message.text)
                .scaledFont(13.5)
                .foregroundStyle(message.fromCline ? DT.ink(scheme) : Color.white)
                .padding(.horizontal, 13).padding(.vertical, 10)
                .background(message.fromCline ? AnyShapeStyle(DT.surface(scheme)) : AnyShapeStyle(DT.heroGradient))
                .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                .overlay(RoundedRectangle(cornerRadius: 15, style: .continuous)
                    .strokeBorder(message.fromCline ? DT.hairline(scheme) : .clear, lineWidth: 0.8))
            if message.fromCline { Spacer(minLength: 40) }
        }
        .frame(maxWidth: .infinity, alignment: message.fromCline ? .leading : .trailing)
    }

    private var draftCard: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Eyebrow("YOUR PROPOSAL")
                Spacer()
                Text(surface)
                    .font(.system(size: 10, weight: .bold))
                    .foregroundStyle(DT.violetText(scheme))
                    .padding(.horizontal, 8).padding(.vertical, 3)
                    .background(DT.violet.opacity(0.10))
                    .clipShape(Capsule())
            }
            Text(draftTitle ?? "")
                .scaledFont(15, .heavy)
                .foregroundStyle(DT.ink(scheme))
            Text(draftSummary)
                .scaledFont(12.5)
                .foregroundStyle(DT.ink78(scheme))
                .lineLimit(6)

            if sent {
                HStack(spacing: 7) {
                    Image(systemName: "checkmark.circle.fill")
                        .foregroundStyle(DT.live(scheme))
                    Text("Sent — track it in Settings → Feedback & experiments.")
                        .scaledFont(12, .semibold)
                        .foregroundStyle(DT.ink78(scheme))
                }
                .padding(.top, 4)
            } else {
                HStack(spacing: 9) {
                    Button { send(trial: false) } label: {
                        Text("Send suggestion")
                            .scaledFont(13, .bold)
                            .foregroundStyle(.white)
                            .frame(maxWidth: .infinity)
                            .frame(height: 40)
                            .background(DT.heroGradient)
                            .clipShape(RoundedRectangle(cornerRadius: DT.rControl, style: .continuous))
                    }
                    .buttonStyle(Pressable())
                    if matchedFlag != nil {
                        Button { send(trial: true) } label: {
                            Text("Try it for a week")
                                .scaledFont(13, .bold)
                                .foregroundStyle(DT.violetText(scheme))
                                .frame(maxWidth: .infinity)
                                .frame(height: 40)
                                .background(DT.violet.opacity(0.10))
                                .clipShape(RoundedRectangle(cornerRadius: DT.rControl, style: .continuous))
                        }
                        .buttonStyle(Pressable())
                    }
                }
                .padding(.top, 6)
                if matchedFlag != nil {
                    Text("This one maps to a built variant (Focus Home) — trying it flips the flag on this device only, for 7 days, revertible any time.")
                        .font(.system(size: 10))
                        .foregroundStyle(DT.ink55(scheme))
                }
            }
        }
        .padding(14)
        .card(radius: DT.rHero)
    }

    private var surfaceChips: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 7) {
                Text("About:")
                    .font(.system(size: 11, weight: .semibold))
                    .foregroundStyle(DT.ink35(scheme))
                ForEach(surfaces, id: \.self) { option in
                    Button { surface = option } label: {
                        Text(option)
                            .scaledFont(12, .bold)
                            .foregroundStyle(surface == option ? .white : DT.ink78(scheme))
                            .padding(.horizontal, 11).padding(.vertical, 6)
                            .background(surface == option ? AnyShapeStyle(DT.violet) : AnyShapeStyle(DT.surface(scheme)))
                            .clipShape(Capsule())
                            .overlay(Capsule().strokeBorder(surface == option ? .clear : DT.hairline(scheme), lineWidth: 0.8))
                    }
                    .buttonStyle(Pressable())
                }
            }
            .padding(.horizontal, 18)
        }
        .padding(.vertical, 9)
    }

    private var inputBar: some View {
        HStack(spacing: 9) {
            TextField("Describe the feature you'd want…", text: $input, axis: .vertical)
                .scaledFont(14)
                .lineLimit(1...3)
                .padding(.horizontal, 13).padding(.vertical, 9)
                .background(DT.surface2(scheme))
                .clipShape(RoundedRectangle(cornerRadius: 17, style: .continuous))
            Button { receive() } label: {
                Image(systemName: "arrow.up")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundStyle(.white)
                    .frame(width: 36, height: 36)
                    .background(input.trimmingCharacters(in: .whitespaces).isEmpty
                                ? AnyShapeStyle(DT.ink35(scheme)) : AnyShapeStyle(DT.heroGradient))
                    .clipShape(Circle())
            }
            .buttonStyle(Pressable())
            .disabled(input.trimmingCharacters(in: .whitespaces).isEmpty)
            .accessibilityLabel("Send message")
        }
        .padding(.horizontal, 16)
        .padding(.bottom, 12)
    }

    private func receive() {
        let text = input.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !text.isEmpty else { return }
        input = ""
        messages.append(FBMessage(fromCline: false, text: text))

        if draftTitle == nil {
            let words = text.split(separator: " ").prefix(7).joined(separator: " ")
            draftTitle = words.prefix(1).uppercased() + words.dropFirst()
            draftSummary = text
            let trialNote = Variants.matches(text, surface: surface) != nil
                ? " This one maps to a variant I can switch on right now if you want to feel it for a week."
                : ""
            messages.append(FBMessage(fromCline: true,
                text: "Got it — here's how I'd pitch it. Pick which screen it's about, add anything, then send.\(trialNote)"))
        } else {
            draftSummary += " " + text
            messages.append(FBMessage(fromCline: true, text: "Folded that in."))
        }
    }

    private func send(trial: Bool) {
        guard let title = draftTitle else { return }
        let flag = matchedFlag
        store.submitSuggestion(title: title, summary: draftSummary, surface: surface, flag: flag)
        if trial, flag != nil, let first = store.experiments.first {
            store.startTrial(first.id)
        }
        sent = true
        UINotificationFeedbackGenerator().notificationOccurred(.success)
        Task {
            try? await Task.sleep(nanoseconds: 1_100_000_000)
            dismiss()
        }
    }
}

// MARK: - Settings section

/// FEEDBACK & EXPERIMENTS — both switches, the experiment list, and the
/// consent gate. Lives inside Configuration settings.
struct FeedbackSettingsSection: View {
    @EnvironmentObject var store: AppStore
    @Environment(\.colorScheme) private var scheme
    @State private var consentSheet = false
    @State private var chatOpen = false

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Eyebrow("FEEDBACK & EXPERIMENTS").padding(.leading, 14).padding(.top, 8)
            VStack(spacing: 0) {
                toggleRow("Feedback program", isOn: Binding(
                    get: { store.feedbackProgramOn },
                    set: { store.feedbackProgramOn = $0 }))
                hairline
                HStack {
                    Text("Feedback mode").scaledFont(15)
                    Spacer()
                    Toggle("Feedback mode", isOn: optInBinding)
                        .labelsHidden()
                        .tint(DT.live(scheme))
                        .disabled(!store.feedbackProgramOn)
                }
                .padding(.horizontal, 14).padding(.vertical, 6).frame(minHeight: 46)

                if store.feedbackAvailable {
                    hairline
                    Button { chatOpen = true } label: {
                        HStack(spacing: 10) {
                            AvatarChip(letter: "C", color: DemoData.coder, size: 24)
                            Text("Suggest something").scaledFont(15).foregroundStyle(DT.violetText(scheme))
                            Spacer()
                            Image(systemName: "bubble.left.and.text.bubble.right")
                                .font(.system(size: 13))
                                .foregroundStyle(DT.ink35(scheme))
                        }
                        .padding(.horizontal, 14).frame(minHeight: 46)
                        .contentShape(Rectangle())
                    }
                    .buttonStyle(.plain)
                }

                ForEach(store.experiments) { experiment in
                    hairline
                    experimentRow(experiment)
                }
            }
            .card()
            .padding(.top, 7)
            Text(store.feedbackProgramOn
                 ? "Two switches: the program (ours) and feedback mode (yours). Trials are look-and-feel only, live 7 days max on this device, and revert any time. Preview build: you hold both switches."
                 : "Program off — feedback UI hidden everywhere, trials reverted, opt-in cleared.")
                .font(.system(size: 11))
                .foregroundStyle(DT.ink55(scheme))
                .padding(.horizontal, 14)
                .padding(.top, 7)
        }
        .sheet(isPresented: $consentSheet) {
            FeedbackPolicyView(consenting: true) {
                store.feedbackOptIn = true
            }
        }
        .sheet(isPresented: $chatOpen) {
            FeedbackChatView()
        }
    }

    /// Turning ON routes through the policy screen; OFF is immediate.
    private var optInBinding: Binding<Bool> {
        Binding(
            get: { store.feedbackOptIn },
            set: { wantsOn in
                if wantsOn { consentSheet = true } else { store.feedbackOptIn = false }
            })
    }

    private func experimentRow(_ experiment: Experiment) -> some View {
        HStack(spacing: 10) {
            Image(systemName: experiment.status == .trialing ? "flask.fill" : "flask")
                .font(.system(size: 13))
                .foregroundStyle(experiment.status == .trialing ? DT.violetText(scheme) : DT.ink35(scheme))
                .frame(width: 24)
            VStack(alignment: .leading, spacing: 2) {
                Text(experiment.title).scaledFont(13.5, .semibold).lineLimit(1)
                Text(experiment.statusLine)
                    .font(.system(size: 10.5))
                    .foregroundStyle(experiment.status == .trialing ? DT.violetText(scheme) : DT.ink55(scheme))
            }
            Spacer()
            if experiment.status == .trialing {
                Button { store.endTrial(experiment.id) } label: {
                    Text("End trial")
                        .scaledFont(11.5, .bold)
                        .foregroundStyle(DT.ink55(scheme))
                        .padding(.horizontal, 10).padding(.vertical, 6)
                        .background(DT.surface2(scheme))
                        .clipShape(Capsule())
                }
                .buttonStyle(Pressable())
            } else if experiment.status != .trialing, !experiment.flag.isEmpty,
                      experiment.status == .suggested {
                Button { store.startTrial(experiment.id) } label: {
                    Text("Try for a week")
                        .scaledFont(11.5, .bold)
                        .foregroundStyle(DT.violetText(scheme))
                        .padding(.horizontal, 10).padding(.vertical, 6)
                        .background(DT.violet.opacity(0.10))
                        .clipShape(Capsule())
                }
                .buttonStyle(Pressable())
            }
        }
        .padding(.horizontal, 14).padding(.vertical, 8).frame(minHeight: 48)
    }

    private var hairline: some View {
        Rectangle().fill(DT.hairline(scheme)).frame(height: 0.8).padding(.leading, 14)
    }

    private func toggleRow(_ label: String, isOn: Binding<Bool>) -> some View {
        HStack {
            Text(label).scaledFont(15)
            Spacer()
            Toggle(label, isOn: isOn).labelsHidden().tint(DT.live(scheme))
        }
        .padding(.horizontal, 14).padding(.vertical, 6).frame(minHeight: 46)
    }
}

File: web/src/views/profile/FeedbackMode.js
// A port of `Sources/FeedbackMode.swift` — the feedback program: a floating
// bubble (only while the program is on AND this device opted in), the design
// chat with Cline (ephemeral, in memory, only the structured suggestion ever
// leaves on an explicit Send), and the Settings section with the two
// switches, the experiments list and the 7-day trial clock.
import { html, cx, useState, useRef, useEffect, useObservable, haptic } from "../../ui.js";
import { NavBar, Icon, AvatarChip, ClineBot, Eyebrow, Card, TextField, Button, ToggleRow, Toggle, Row } from "../../components.js";
import { agentColor } from "../../models.js";
import { ctx, SectionLabel, Footnote, Hairline } from "./shared.js";

// ------------------------------------------------------------- model

export const ExperimentStatus = { suggested: "suggested", trialing: "trialing", reverted: "reverted", expired: "expired" };

/** Days left on a trial (nil when not trialing). */
export function daysLeft(e, now = Date.now()) {
  if (e.status !== "trialing" || e.expiresAt == null) return null;
  return Math.max(0, Math.ceil((e.expiresAt - now) / 86_400_000));
}

export function statusLine(e) {
  switch (e.status) {
    case "suggested": return "Suggested · awaiting decision";
    case "trialing": { const d = daysLeft(e); return d == null ? "Trial" : `Trial · ${d}d left on this device`; }
    case "reverted": return "Trial ended by you · still in review";
    case "expired": return "Trial reached its week · still in review";
    default: return "";
  }
}

/** The variants this build knows how to try. Presentation-only, always. */
export const Variants = {
  focusHome: "focus-home",
  matches(text, surface) {
    const folded = String(text ?? "").toLowerCase();
    const focusWords = ["focus", "minimal", "simpl", "hide", "quiet", "clean", "less"];
    if ((surface === "Home" || folded.includes("home")) && focusWords.some((w) => folded.includes(w))) return Variants.focusHome;
    return null;
  },
};

/** The one variant the preview ships (docs/FEEDBACK-MODE.md): Focus Home,
 *  suggested and ready to trial. Production seeds nothing. */
const FOCUS_HOME_SEED = {
  id: "exp-focus-home", title: "Focus Home", detail: "Hide Home's ARTIFACTS and RECENT sections so the floor stays quiet while agents work.",
  surface: "Home", flag: Variants.focusHome, status: "suggested", startedAt: null, expiresAt: null, sentAt: 0,
};
export function seedExperiments(store) {
  if (store.experiments != null) return;
  store.setExperiments(store.configuration.feedbackExperimentsEnabled ? [FOCUS_HOME_SEED] : []);
}

// ----------------------------------------------------- floating bubble

/** The feedback door — only exists while the program is on AND this device
 *  opted in. Quiet, corner, never over the session plane. The Swift app mounts
 *  it on Home (HomeView.swift) while not in a call; here it rides Profile. */
export function FeedbackBubble() {
  const store = useObservable(ctx.store);
  if (!store.feedbackAvailable || store.inCall) return null;
  return html`<button type="button" class="pf-bubble pressable" aria-label="Design with Cline" title="Suggest a feature — chat stays on this device" onClick=${() => ctx.nav.present("feedbackChat")}>
    <${ClineBot} size=${22} color="#fff" />
  </button>`;
}

// ------------------------------------------------------- design chat

const SURFACES = ["Home", "Work", "Agents", "Tasks", "Profile", "Artifacts"];
let msgCounter = 0;
const msg = (fromCline, text) => ({ id: `fb-${++msgCounter}`, fromCline, text });

/** Chat with Cline to shape a suggestion. The conversation is ephemeral —
 *  in memory, gone when the sheet closes. Only the structured draft leaves,
 *  and only on an explicit Send. (Cline here is a rule-based design persona
 *  in the preview build; the honesty chrome says so.) */
export function FeedbackChatView() {
  const store = useObservable(ctx.store);
  const [messages, setMessages] = useState(() => [msg(true, "What should Drive do better? Describe it like you'd tell a friend — I'll shape it into a proposal you can send.")]);
  const [input, setInput] = useState("");
  const [surface, setSurface] = useState("Home");
  const [draftTitle, setDraftTitle] = useState(null);
  const [draftSummary, setDraftSummary] = useState("");
  const [sent, setSent] = useState(false);
  const scrollRef = useRef();
  const matchedFlag = Variants.matches(draftSummary, surface);

  useEffect(() => { const el = scrollRef.current; if (el) el.scrollTop = el.scrollHeight; }, [messages.length, draftTitle, sent]);

  const receive = () => {
    const text = input.trim();
    if (!text) return;
    setInput("");
    const next = [...messages, msg(false, text)];
    if (draftTitle == null) {
      const words = text.split(/\s+/).slice(0, 7).join(" ");
      setDraftTitle(words.charAt(0).toUpperCase() + words.slice(1));
      setDraftSummary(text);
      const trialNote = Variants.matches(text, surface) ? " This one maps to a variant I can switch on right now if you want to feel it for a week." : "";
      next.push(msg(true, `Got it — here's how I'd pitch it. Pick which screen it's about, add anything, then send.${trialNote}`));
    } else {
      setDraftSummary((s) => `${s} ${text}`);
      next.push(msg(true, "Folded that in."));
    }
    setMessages(next);
  };

  const send = (trial) => {
    if (!draftTitle) return;
    const flag = matchedFlag;
    store.submitSuggestion({ title: draftTitle, summary: draftSummary, surface, flag });
    if (trial && flag && store.experiments?.[0]) store.startTrial(store.experiments[0].id);
    setSent(true);
    haptic("success");
    setTimeout(() => ctx.nav.dismiss(), 1100);
  };

  return html`<div class="pf-chat">
    <${NavBar} title="Design with Cline" leading=${html`<${Button} variant="ghost" size="sm" onClick=${() => ctx.nav.dismiss()}>Close</${Button}>`} />
    <div class="pf-honesty"><i class="dot" />CHAT STAYS ON THIS DEVICE · ONLY WHAT YOU SEND LEAVES</div>
    <div class="pf-msgs" ref=${scrollRef} aria-live="polite">
      ${messages.map((m) => html`<div key=${m.id} class=${cx("pf-msg", !m.fromCline && "me")}>
        ${m.fromCline ? html`<${AvatarChip} letter="C" name="Cline" color=${agentColor("coder")} size=${24} />` : null}
        <div class="pf-mbody">${m.text}</div>
      </div>`)}
      ${draftTitle != null ? html`<${Card} hero className="pf-draft pop-in" style=${{ marginTop: 6 }}>
        <div class="hstack" style=${{ justifyContent: "space-between" }}><${Eyebrow}>YOUR PROPOSAL</${Eyebrow}><span class="pf-dsurf">${surface}</span></div>
        <div class="pf-dtitle" style=${{ marginTop: 8 }}>${draftTitle}</div>
        <div class="pf-dsum" style=${{ marginTop: 6 }}>${draftSummary}</div>
        ${sent ? html`<div class="hstack" style=${{ gap: 7, marginTop: 10 }}><${Icon} name="checkmark.circle.fill" size=${16} fill color="var(--live)" /><span class="t-sm w6 ink78">Sent — track it in Settings → Feedback & experiments.</span></div>`
        : html`<div>
          <div class="hstack" style=${{ gap: 9, marginTop: 12 }}>
            <${Button} variant="gradient" fill onClick=${() => send(false)} style=${{ minHeight: 40, fontSize: 13 }}>Send suggestion</${Button}>
            ${matchedFlag ? html`<${Button} fill onClick=${() => send(true)} style=${{ minHeight: 40, fontSize: 13, color: "var(--violet-text)", background: "color-mix(in srgb, var(--violet) 10%, transparent)", boxShadow: "none" }}>Try it for a week</${Button}>` : null}
          </div>
          ${matchedFlag ? html`<div class="t-xs muted" style=${{ marginTop: 8, fontSize: 10, lineHeight: 1.4 }}>This one maps to a built variant (Focus Home) — trying it flips the flag on this device only, for 7 days, revertible any time.</div>` : null}
        </div>`}
      </${Card}>` : null}
    </div>
    <div class="pf-chips" role="radiogroup" aria-label="Which screen this is about">
      <span class="pf-about">About:</span>
      ${SURFACES.map((s) => html`<button key=${s} type="button" role="radio" aria-checked=${surface === s} class=${cx(surface === s && "on")} onClick=${() => { haptic("light"); setSurface(s); }}>${s}</button>`)}
    </div>
    <div class="pf-inputbar">
      <${TextField} value=${input} onInput=${setInput} placeholder="Describe the feature you'd want…" multiline rows=${1} clearable=${false} onSubmit=${receive} label="Message" autoFocus />
      <button type="button" class="pf-send pressable" aria-label="Send message" disabled=${!input.trim()} onClick=${receive}><${Icon} name="arrow.up" size=${16} weight=${3} /></button>
    </div>
  </div>`;
}

// ---------------------------------------------------- settings section

/** FEEDBACK & EXPERIMENTS — both switches, the experiment list, and the
 *  consent gate. Lives inside Configuration settings. */
export function FeedbackSettingsSection() {
  const store = useObservable(ctx.store);
  const experiments = store.experiments ?? [];
  const setOptIn = (wantsOn) => {
    // Turning ON routes through the policy screen; OFF is immediate.
    if (wantsOn) ctx.nav.present("policy", { kind: "feedback", consenting: true });
    else store.setFeedbackOptIn(false);
  };
  return html`<div>
    <${SectionLabel}>FEEDBACK & EXPERIMENTS</${SectionLabel}>
    <div class="card" style=${{ overflow: "hidden", padding: 0 }}>
      <${ToggleRow} title="Feedback program" checked=${store.feedbackProgramOn} onChange=${(v) => store.setFeedbackProgramOn(v)} />
      <${Row} title="Feedback mode" trailing=${html`<${Toggle} checked=${store.feedbackOptIn} onChange=${setOptIn} label="Feedback mode" disabled=${!store.feedbackProgramOn} />`} />
      ${store.feedbackAvailable ? html`<${Row} leading=${html`<${AvatarChip} letter="C" name="Cline" color=${agentColor("coder")} size=${24} />`} title=${html`<span class="violet" style=${{ fontWeight: 500 }}>Suggest something</span>`} trailing=${html`<${Icon} name="bubble.left.and.text.bubble.right" size=${14} color="var(--ink-35)" />`} onClick=${() => ctx.nav.present("feedbackChat")} label="Suggest something" />` : null}
      ${experiments.map((e) => html`<${ExperimentRow} key=${e.id} experiment=${e} store=${store} />`)}
    </div>
    <${Footnote}>${store.feedbackProgramOn
      ? "Two switches: the program (ours) and feedback mode (yours). Trials are look-and-feel only, live 7 days max on this device, and revert any time. Preview build: you hold both switches."
      : "Program off — feedback UI hidden everywhere, trials reverted, opt-in cleared."}</${Footnote}>
  </div>`;
}

function ExperimentRow({ experiment: e, store }) {
  const trialing = e.status === "trialing";
  return html`<div>
    <${Hairline} />
    <div class=${cx("pf-exp", trialing && "trialing")}>
      <span class="pf-exp-icon"><${Icon} name="flask.fill" size=${14} weight=${2.2} fill=${trialing} /></span>
      <div class="grow">
        <div class="pf-exp-title">${e.title}</div>
        <div class="pf-exp-status">${statusLine(e)}</div>
      </div>
      ${trialing ? html`<button type="button" class="pf-pill-btn pressable" onClick=${() => { haptic("light"); store.endTrial(e.id); }}>End trial</button>`
      : e.status === "suggested" && e.flag ? html`<button type="button" class="pf-pill-btn violet pressable" onClick=${() => { haptic("light"); store.startTrial(e.id); }}>Try for a week</button>` : null}
    </div>
  </div>`;
}
